Adding most of the inter server tests back

lumarel opened the pull request:
# Description

This pull request will add the inter-server tests back into the `templates.fif.json` and `templates-updates.fif.json` files.
Further there will be added some small fixes which occurred while adding. (basically small script fixes and additional needles)
To finish, will also update the docs so everything is correct about these tests

This PR fixes #73 

# How Has This Been Tested?

```bash
openqa-cli api -X POST isos ISO=Rocky-8.5-x86_64-dvd1.iso ARCH=x86_64 DISTRI=rocky FLAVOR=dvd-iso PACKAGE_SET=server VERSION=8.5 CURRREL=8.5 LOCATION=https://dl.rockylinux.org/pub/rocky/8.5 BUILD=-server-$(date +%Y%m%d.%H%M%S).0
openqa-cli api -X POST isos ISO=Rocky-8.5-x86_64-dvd1.iso ARCH=x86_64 DISTRI=rocky FLAVOR=universal VERSION=8.5 CURRREL=8.5 LOCATION=https://dl.rockylinux.org/pub/rocky/8.5 BUILD=-universal-$(date +%Y%m%d.%H%M%S).0
```

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

tcooper commented:
@lumarel Looking at your commit messages I can see I was confused with what you were saying. I'll look at merging my changes with yours and verify the `targetcli` stuff in my testing environment.

I have `install_pxe@x86_54` working aside from figuring out how to make the bootloader menu stay on screen after reboot long enough for a needle to be captured. I may have been successful creating one from a screen grab of the on-screen Video but am running it through again now to check.

tcooper commented:
@lumarel This might get you further along with `_support_server.pm`...

```diff
diff --git a/tests/_support_server.pm b/tests/_support_server.pm
index 5ae283c1..23dc742e 100644
--- a/tests/_support_server.pm
+++ b/tests/_support_server.pm
@@ -8,6 +8,9 @@ use tapnet;
 use utils;

 sub _pxe_setup {
+    # This happens before local DNS server is running and dnf will fail. Temporarily add 8.8.8.8 to resolve.conf
+    assert_script_run "echo 'nameserver 8.8.8.8' >> /etc/resolv.conf";
+
     # set up PXE server (via dnsmasq). Not used for update tests.
     # don't get hung up on slow mirrors when DNFing...
     repos_mirrorlist;
@@ -22,6 +25,7 @@ sub _pxe_setup {
     assert_script_run "printf 'dhcp-match=set:efi-x86_64,option:client-arch,7\ndhcp-match=set:efi-x86_64,option:client-arch,9\ndhcp-match=set:bios,option:client-arch,0\ndhcp-match=set:efi-aarch64,option:client-arch,11\ndhcp-match=set:ppc64,option:client-arch,12\ndhcp-match=set:ppc64,option:client-arch,13\ndhcp-boot=tag:efi-x86_64,\"shim.efi\"\ndhcp-boot=tag:bios,\"pxelinux.0\"\ndhcp-boot=tag:efi-aarch64,\"grubaa64.efi\"\ndhcp-boot=tag:ppc64,\"boot/grub2/powerpc-ieee1275/core.elf\"\n' >> /etc/dnsmasq.conf";
     # install and configure bootloaders
     my $ourversion = get_var("CURRREL");
+    my $ourversion = "8";
     my $testversion = get_var("RELEASE");
     assert_script_run "mkdir -p /var/tmp/rocky";
     my $arch = get_var("ARCH");
@@ -31,15 +35,23 @@ sub _pxe_setup {
         assert_script_run "mkdir -p /var/lib/tftpboot/pxelinux.cfg";
         # install bootloader packages
         assert_script_run "dnf -y install syslinux", 120;
-        assert_script_run "dnf -y --releasever=$ourversion --installroot=/var/tmp/rocky install shim-x64 grub2-efi-x64", 300;
+        assert_script_run "rpm --root=/var/tmp/rocky --rebuilddb", 60;
+        assert_script_run "cd /var/tmp; dnf download rocky-release rocky-repos rocky-gpg-keys", 60;
+        assert_script_run "rpm --root=/var/tmp/rocky --nodeps -i /var/tmp/*.rpm", 60;
+        #assert_script_run "sed -i /var/tmp/rocky/etc/yum.repos.d/Rocky-*.repo -e 's/repo=/repo=rocky-/g'", 60;
+        assert_script_run "dnf -y --releasever=$ourversion --refresh --installroot=/var/tmp/rocky install shim-x64 grub2-efi-x64", 1800;
+
         # copy bootloader files to tftp root
         assert_script_run "cp /usr/share/syslinux/{pxelinux.0,vesamenu.c32,ldlinux.c32,libcom32.c32,libutil.c32} /var/lib/tftpboot";
-        assert_script_run "cp /var/tmp/rocky/boot/efi/EFI/rocky/{shim.efi,grubx64.efi} /var/lib/tftpboot";
+        assert_script_run "cp /var/tmp/rocky/boot/efi/EFI/rocky/{shimx64.efi,shimx64-rocky.efi,grubx64.efi} /var/lib/tftpboot";
         # bootloader configs
         # BIOS
-        assert_script_run "printf 'default vesamenu.c32\nprompt 1\ntimeout 600\n\nlabel linux\n  menu label ^Install Rocky Linux\n  menu default\n  kernel rocky/vmlinuz\n  append initrd=rocky/initrd.img inst.ks=file:///ks.cfg ip=dhcp\nlabel local\n  menu label Boot from ^local drive\n  localboot 0xffff\n' >> /var/lib/tftpboot/pxelinux.cfg/default";
+        assert_script_run "printf 'default vesamenu.c32\nprompt 1\ntimeout 600\n\nlabel linux\n  menu label ^Install Rocky Linux 64-bit\n  menu default\n  kernel rocky/vmlinuz\n  append initr
+d=rocky/initrd.img inst.ks=file:///ks.cfg ip=dhcp\nlabel local\n  menu label Boot from ^local drive\n  localboot 0xffff\n' >> /var/lib/tftpboot/pxelinux.cfg/default";
         # UEFI
-        assert_script_run "printf 'function load_video {\n  insmod efi_gop\n  insmod efi_uga\n  insmod ieee1275_fb\n  insmod vbe\n  insmod vga\n  insmod video_bochs\n  insmod video_cirrus\n}\n\nload_video\nset gfxpayload=keep\ninsmod gzio\n\nmenuentry \"Install Rocky Linux\"  --class rocky --class gnu-linux --class gnu --class os {\n  linuxefi rocky/vmlinuz ip=dhcp inst.ks=file:///ks.cfg\n  initrdefi rocky/initrd.img\n}' >> /var/lib/tftpboot/grub.cfg";
+        assert_script_run "printf 'function load_video {\n  insmod efi_gop\n  insmod efi_uga\n  insmod ieee1275_fb\n  insmod vbe\n  insmod vga\n  insmod video_bochs\n  insmod video_cirrus\n}\
+n\nload_video\nset gfxpayload=keep\ninsmod gzio\n\nmenuentry \"Install Rocky Linux 64-bit\"  --class rocky --class gnu-linux --class gnu --class os {\n  linuxefi rocky/vmlinuz ip=dhcp inst.ks=
+file:///ks.cfg\n  initrdefi rocky/initrd.img\n}' >> /var/lib/tftpboot/grub.cfg";
         # DEBUG DEBUG
         upload_logs "/etc/dnsmasq.conf";
         upload_logs "/var/lib/tftpboot/grub.cfg";
@@ -49,7 +61,7 @@ sub _pxe_setup {
     elsif ($arch eq 'ppc64le') {
         # ppc64le: use grub2 for OFW
         # install bootloader tools package
-        assert_script_run "dnf -y install grub2-tools-extra", 180;
+        assert_script_run "dnf -y install grub2-tools-extra", 360;
         # install a network bootloader to tftp root
         assert_script_run "grub2-mknetdir --net-directory=/var/lib/tftpboot";
         # bootloader config
@@ -80,7 +92,8 @@ sub _pxe_setup {
     assert_script_run "curl -o /var/lib/tftpboot/rocky/initrd.img $location/BaseOS/${arch}/os/${kernpath}/initrd.img";
     # get a kickstart to embed in the initramfs, for testing:
     # https://fedoraproject.org/wiki/QA:Testcase_Kickstart_File_Path_Ks_Cfg
-    assert_script_run "curl -o ks.cfg https://fedorapeople.org/groups/qa/kickstarts/root-user-crypted-net.ks";
+    #assert_script_run "curl -o ks.cfg https://fedorapeople.org/groups/qa/kickstarts/root-user-crypted-net.ks";
+    assert_script_run "curl -o ks.cfg https://git.rockylinux.org/tcooper/kickstarts/-/raw/fc7366d2668db5e52a921a86f13f03da5897217a/root-user-crypted-net.ks";
     # tweak the repo config in it
     assert_script_run "sed -i -e 's,^url.*,nfs --server 172.16.2.110 --dir /repo --opts nfsvers=4,g' ks.cfg";
     # embed it
@@ -101,14 +114,17 @@ sub run {
         script_run "rm -f /etc/resolv.conf";
         script_run "systemctl restart NetworkManager";
     }
+
     ## DNS / DHCP (dnsmasq)
     # create config
-    assert_script_run "printf 'domain=test.openqa.fedoraproject.org\ndhcp-range=172.16.2.150,172.16.2.199\ndhcp-option=option:router,172.16.2.2\n' > /etc/dnsmasq.conf";
+    assert_script_run "printf 'domain=test.openqa.rockylinux.org\ndhcp-range=172.16.2.150,172.16.2.199\ndhcp-option=option:router,172.16.2.2\n' > /etc/dnsmasq.conf";
     # do PXE setup if this is not an update test
     _pxe_setup() unless (get_var("ADVISORY_OR_TASK"));
+
     # open firewall ports
     assert_script_run "firewall-cmd --add-service=dhcp";
     assert_script_run "firewall-cmd --add-service=dns";
+
     # start server
     assert_script_run "systemctl restart dnsmasq.service";
     assert_script_run "systemctl is-active dnsmasq.service";
@@ -131,7 +147,7 @@ sub run {
     # create the file share
     assert_script_run "mkdir -p /export";
     # get the kickstart
-    assert_script_run "curl -o /export/root-user-crypted-net.ks https://fedorapeople.org/groups/qa/kickstarts/root-user-crypted-net.ks";
+    assert_script_run "curl -o /export/root-user-crypted-net.ks https://git.rockylinux.org/tcooper/kickstarts/-/raw/fc7366d2668db5e52a921a86f13f03da5897217a/root-user-crypted-net.ks";
     # for update tests, set up the update repository and export it
     if (get_var("ADVISORY_OR_TASK")) {
         assert_script_run "echo '/opt/update_repo 172.16.2.0/24(ro)' >> /etc/exports";
@@ -146,7 +162,7 @@ sub run {
         assert_script_run "mount /dev/cdrom /mnt/iso";
         # copy the contents of the ISO to the repo share
         assert_script_run "dnf -y install rsync", 180;
-        assert_script_run "rsync -av /mnt/iso/ /repo", 180;
+        assert_script_run "rsync -av /mnt/iso/ /repo", 360;
         # put the updates image in the NFS repo (for testing this update
         # image delivery method)
         assert_script_run "curl -o /repo/images/updates.img https://fedorapeople.org/groups/qa/updates/updates-openqa.img";
```

We can talk about it on Thursday.

tcooper commented:
@lumarel I may also have the `install_scsi` test fixed for `targetcli`. If you haven't figured that out maybe wait until you see what I've got tomorrow (later today your time).

tcooper commented:
@lumarel give this ☝🏻 a try.

lumarel commented:
Looks like it should be there for now!
If you see any further changes which should be made, I will off course retest and change 👌🏻

tcooper commented on needles/anaconda/install_destination/rocky-add_network_disk-20220217.json:
```diff
@@ -0,0 +1,15 @@
+  "properties": [],
+  "tags": [
+    "anaconda_install_destination_add_network_disk"
+  ]
```
Needs `ENV-DISTRI-rocky`?

tcooper commented on needles/anaconda/partitioning/rocky-device_sda_selected-20220217.json:
```diff
@@ -0,0 +1,16 @@
+  "tags": [
+    "anaconda_part_device_sda",
+    "anaconda_part_device_sda_selected"
+  ]
```
Needs `ENV-DISTRI-rocky`

tcooper reviewed (changes requested):
Like @akatch I don't want to block this anymore. I used to have this working in my systems but have yet to restore tap device functionality. I noted a few needles json where `ENV-DISTRI-rocky` appears to be missing... is this intentional? If it is then conversation can be resolved and I will change to approve. Otherwise, needs work.

tcooper commented:
Also, there is/are now merge conflicts for `templates.fif.json`. Maybe push a commit that resolves those as well or rebase branch.

lumarel commented:
Okay sorry for the long wait, I rebased on `develop` now and fixed the 2 needles where the `ENV-DISTRI-rocky` was missing, if the time is now, i will start some full runs, and check over the code once again!
I'm open for any further changes of course!

akatch reviewed (approved):
Thanks Lukas!

tcooper commented:
@lumarel While I said 3 weeks ago I didn't want to block this it looks like I did just that. Completed my review which unblocks but I feel like you may have worked on this some more since them. Is that in another branch and not pushed here? I'm thinking these tests were for Rocky 8 and you've been working on inter-server for Rocky 9. If this is good to go please go ahead and merge otherwise let us know if/what is needed from us. Sorry for the delay...

tcooper commented:
@lumarel just glancing at this again after looking at updates for createhdds and see a few changes needed for the Rocky 8 vs 9 diffs. One example is that `/etc/yum.repos.d/*` filenames are different. Maybe it would be good to drop a few commits onto this PR to take care of that. Also, we'll want to very clearly document the `$CURRREL` variable setting as this is tied to the `createhdds` repo values for `release` inside `hdds.json`. The two repos need to be in lockstep here.

lumarel commented:
This was just the rebase today, will update and test the rest over the weekend (hopefully)

tcooper commented:
@akatch Since @StackKorora cannot run/review can you redo your review and approve so that this can be merged?

tcooper commented:
Reviewing the discussion in this PR I can see that others are not able to reproduce results due to environment issues. Only @AlanMarshall is missing and I'm going to add as a reviewer to see if he can run through tests of this PR to get us to two reviews and approvals. If that isn't possible I propose we run a copy of this PR through in our production instance in order to get confirmation in that environment it works. If that is successful I would be amenable to merge with only my review and with @lumarel and prod confirmation. If anybody objects please speak up...

AlanMarshall commented:
@tcooper  I have been thinking about this for some time with a view to how I could do a review and preferably approve this PR.  I don't have multi-vm available yet on any of my equipment so I can't positively test this PR locally.
However,  I think that this is a special case because of the effective split in the way that people actually install and use openqa.  As you know, the initial install is relatively straightforward using single vm's to carry out most of the testing that is required and most people stay at this stage for a relatively long time before attempting to add multi-vm capability.  This means that a sizeable portion of the tester population use single vm's and any multi-vm tests that are run by them will fail.
It seems to me therefore that a test to check that single vm tests on a system without multi-vm capability are not adversely affected by this PR would be especially valid.  I intended to propose to do this in order to issue a qualified approval,  unless there is an objection from others.
If I could also initiate, or look at the results of,  a positive confirming test on the production instance then I could give a full approval on the above basis.

AlanMarshall reviewed (approved):
This is a qualified approval.
I would not normally give an approving review of a PR that I had not positively tested, however as noted above, this is a special case.  The consensus in the test team is that in order to move on, this PR should be merged and any residual problems sorted out later.  As noted above in an earlier approving review, positive confirming tests have been carried out.  This present set of tests is a check that systems without multi-vm capability will not be adversely affected by PR#74 when it is merged into develop branch.

This was tested by running branch feature/inter-server-tests-all-open-prs:
```
openqa-cli api -X POST isos ISO=Rocky-9.1-20221214.1-x86_64-dvd.iso ARCH=x86_64 DISTRI=rocky FLAVOR=package-set VERSION=9.1 BUILD=-"$(date +%Y%m%d.%H%M%S).0"-9.1-20221214.1-package-set-#74+dev
openqa-cli api -X POST isos ISO=Rocky-9.1-20221214.1-x86_64-dvd.iso ARCH=x86_64 DISTRI=rocky FLAVOR=dvd-iso VERSION=9.1 BUILD=-"$(date +%Y%m%d.%H%M%S).0"-9.1-20221214.1-dvd-iso-#74+dev
openqa-cli api -X POST isos ISO=Rocky-9.1-20221214.1-x86_64-dvd.iso ARCH=x86_64 DISTRI=rocky FLAVOR=universal VERSION=9.1 BUILD=-"$(date +%Y%m%d.%H%M%S).0"-9.1-20221214.1-universal-#74+dev

openqa-cli api -X POST isos ISO=Rocky-8.7-x86_64-dvd1.iso ARCH=x86_64 DISTRI=rocky FLAVOR=dvd-iso VERSION=8.7 BUILD=-"$(date +%Y%m%d.%H%M%S).0"-8.7-20221110-dvd-iso-#74+dev
openqa-cli api -X POST isos ISO=Rocky-8.7-x86_64-dvd1.iso ARCH=x86_64 DISTRI=rocky FLAVOR=universal VERSION=8.7 BUILD=-"$(date +%Y%m%d.%H%M%S).0"-8.7-20221110-universal-#74+dev
openqa-cli api -X POST isos ISO=Rocky-8.7-x86_64-dvd1.iso ARCH=x86_64 DISTRI=rocky FLAVOR=package-set VERSION=8.7 BUILD=-"$(date +%Y%m%d.%H%M%S).0"-8.7-20221110-package-set-#74+dev
```
The package-set flavor tests are not affected. The universal flavor tests have 3 extra tests which are skipped because they are associated with multi-vm.  On 9.1 the dvd-iso flavor tests have 20 extra tests which are similarly skipped. On 8.7 the dvd-iso flavor has 1 missing needle that takes out 2 dependencies, otherwise same as 9.1.    Missing needles are not affected by multi-vm capability.
All other previous test results as before.

This proves that the PR#74 does not adversely affect tests run on systems that do not have multi-vm capability.
